### DISCUSSION TITLE
Welcome to whetstone.chatgpt Discussions!

### DISCUSSION THREAD

**Discussion by johniwasz** [johniwasz/whetstone.chatgpt#8] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by csocha2** [comment]
Hey there, 
  I am using your library. It's great and easy to use! Nice work. One quick question...are there any plans to support "SessionId" in the ChatGPTCompletionRequest object?

**Reply by johniwasz** [reply]
Thanks for the kind words!

So, despite the name this API uses the GPT-3 API which does not include a sessionId when requesting a completion. The GPT-3 completion request is documented in detail [here](https://platform.openai.com/docs/api-reference/completions/create).

While the [ChatGPT preview](https://chat.openai.com) does use a sessionId, the API that OpenAI uses is not available for use by developers. I haven't seen it published anywhere. The libraries that claim to be using ChatGPT directly are using something like [puppeteer](https://pptr.dev/) to get around CloudFlare. Anything built on this model will stop working when OpenAI takes away their free preview of ChatGPT.

It's possible that OpenAI made a ChatGPT endpoint available from their beta program which requires a $40/mo. subscription fee.  They mention a ChatGPT API waitlist on this [page](https://platform.openai.com/overview). 

![image](https://user-images.githubusercontent.com/44126651/216829981-6e690c4d-888e-4e42-acf1-716be5ce3f0a.png)

I have signed up, but I'm still on the waitlist. When this does become available and I do get access, I will update the API to support it. 

I hope this answers your question?
